src/ipc/ipc4/handler.c

```diff
@@ -510,13 +510,12 @@ static void ipc_compound_msg_done(uint32_t msg_id, int error)
 	}
 }
 
-/* wait for IPCs to complete on other cores and be nice to any LL work */
 static int ipc_wait_for_compound_msg(void)
 {
-	int try_count = 30; /* timeout out is 30 x 10ms so 300ms for IPC */
+	int try_count = 30;
 
 	while (atomic_read(&msg_data.delayed_reply)) {
-		k_sleep(Z_TIMEOUT_MS(10));
+		k_sleep(Z_TIMEOUT_US(250));
```

Comment:
btian1 

250 --> 10ms? it is 40 times enlarge? may miss something during sleep? how about 1ms?

above is the comments I added when the change first submitted, my question is: how 250us * 30 = 7.5ms comes from?

is it based on experience value to make total wait time < 10ms?

Reply:
it is set by experiments with test and log. Most waiting time is less than 1ms.  I don't want to use 1ms since it will miss one schedule period in FW.

Reply:
What context is this called from ? My expectation is the IPC thread and if so not sure how we can block LL from running ?

Reply:
@lgirdwood This is the IPC thread. It doesn't block the LL thread, but it will block processing of IPC messages until the LL timer fires next time, runs the IPC processing (that needs to happen in LL context), acks it as done, and then the IPC thread can continue. Now this adds a mandatory 10ms delay to each such handover from IPC thread to LL thread.

Reply:
ok, so we have a timing dependency between LL thread and IPC thread and this is convergence. This need to be spelled out in the inline comments alongside setting a better tries count as 30 means about 7.5 LL ticks. i.e. we need to say > 1.5 LL ticks would be a significant failure

Reply:
@kv2019i , @RanderWang  is it better if 150 * 50? to make IPC response more quick?

Reply:
@lgirdwood This was the mechanism added by @lyakh to synchronize pipeline trigger processing with LL thread (so that pipeline state changes are not done async with the pipeline exection). @lgirdwood @btian1 We can improve the logic (or even devise a different way to wait for next ll tick, I think the original concern of trying to avoid unnecessary thread context switches is as valid as ever), but can we do that as a follow-up and merge this plain revert first? This helps with multiple pressing issues (AV sync and this seems to help with multicore scenarios), and we have done significant testing of this version with @RanderWang . If we want a modified new mechanism, that needs to be tested well, pushing the merge out.

Reply:
I'm fine with a revert, but we need a folloup PR that spells out the relationships with some inline commentary.